### PR TITLE
Setup Typescript and rename all files to typescript

### PR DESCRIPTION
**Pull request by shrinathprabhu**
# Pull Request Template

Resolves [AR-1929](https://team-1624093970686.atlassian.net/browse/AR-1929).

## Changes
1. Installed `typescript` and `vue-tsc` modules (Ignore `package-lock.json` changes created because of this)
2. Created `env.d.ts` so typescript can understand vue modules
3. Renamed all the js files to ts files (except `postcss.config.js`)
4. Added `lang="ts"` to all the vue component scripts
5. Ingore the changes added because of code editor on save (In vue components)

## Checklist

- [x] The branch name follows the format: `developer/AR-XXX-issue-name`.
- [x] The changes have been tested locally.

### CODE REVIEW

**Comment by ajithranka** [comment]
@shrinathprabhu Are you moving this out of draft?

**Comment by shrinathprabhu** [comment]
> @shrinathprabhu Are you moving this out of draft?

Yes

File: tsconfig.vite-config.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
```
Comment (ajithranka) [inline tsconfig.vite-config.json:1]: Just a note: The formatting here should get fixed once you set up Prettier.

**Review by ajithranka — APPROVED** [review]

File: tsconfig.vite-config.json (hunk 1 above)
Comment (shrinathprabhu) [reply tsconfig.vite-config.json:1]: Sure

**Review by karthik-durai — APPROVED** [review]

**Pull request merged**